samples/csharp_dotnetcore/05.multi-turn-prompt/Bots/DialogBasedBot.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public class DialogBasedBot : ActivityHandler
+    {
+        private ConversationState _conversationState;
+        private UserState _userState;
+        private RootDialog _dialog;
+        private ILogger<DialogBasedBot> _logger;
+
+        public DialogBasedBot(ConversationState conversationState, UserState userState, RootDialog dialog, ILogger<DialogBasedBot> logger)
+        {
+            _conversationState = conversationState;
+            _userState = userState;
+            _dialog = dialog;
+            _logger = logger;
+        }
+
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var dialogSet = new DialogSet(_conversationState.CreateProperty<DialogState>("DialogState"));
+            dialogSet.Add(_dialog);
+
+            var dialogContext = await dialogSet.CreateContextAsync(turnContext, cancellationToken);
```

Comment:
@johnataylor  can you bring the 'Run' method from sample 42 (or 30?)

Reply:
We discussed just this @Kaiqb was in favor of keeping things flat. Perhaps we can iterate a little on this. The big step here is to move all the dialog logic into a ComponentDialog and inject the resulting class. That is what 42 does too. However, 42 also uses a much more straight forward state model. We tried that approach but then found we wanted to use the UserState for parity with the existing sample.